### PR TITLE
Stupns Homework

### PR DESCRIPTION
**Pull request by stupns**

### CODE REVIEW

**Comment by stupns** [comment]
Rewrite

**Comment by stupns** [comment]
end

File: homework/Stupns/homework6.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+url = 'https://raw.githubusercontent.com/aodarc/LITS009/master/lections/4.recurtion_example.py'
+
+
+def download(url):
+    new = requests.get(url, allow_redirects=True)
+    if new.status_code == 404:
+        print('No such file found at %s' % url)
+        return
+    if not url.rfind('.py'):
+        a = print('File extension with .py not found')
```
Comment (aodarc) [inline homework/Stupns/homework6.py]: in this case if you want return None, you should skip `a` variable and use just `return`

File: homework/Stupns/homework9.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import timeit
+
+
+def numberGenerator(n):
```
Comment (aodarc) [inline homework/Stupns/homework9.py]: It should generate `simple numbers` !!!

File: homework/Stupns/homework9.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    number = 0
+    while number < n:
+        yield number
+        number += 1
+
+
+def simple_func():
+    a = []
+    b = a.append(0)
```
Comment (aodarc) [inline homework/Stupns/homework9.py]: Do you like this code? Two mistakes:
1) You do not need create new variable to append something into list. `.append()` method always returns None
2) Why you didn't use cycle?

File: homework/Stupns/homework9.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    g = a.append(4)
+    h = a.append(5)
+    j = a.append(6)
+    k = a.append(8)
+    l = a.append(9)
+    return a
+
+
+if __name__ == '__main__':
+    print('Work Number generator func ~ms :')
```
Comment (aodarc) [inline homework/Stupns/homework9.py]: In this homework you should compare two ways to create lists of simple numbers by time. But you didn't .

**Review by aodarc — CHANGES_REQUESTED** [review]

File: homework/Stupns/homework10.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        def wrapper(*args2):
+            try:
+                if name2 == '':
+                    new_name = str(name).replace('<', '</')
+                    return name + func(*args2) + new_name
+                else:
+                    return name + func(*args2) + name2
+            except TypeError:
+                err = 'You input incorrect *args, plz write one or two args of type str in @'
+                err += tag.__name__
```
Comment (aodarc) [inline homework/Stupns/homework10.py]: use string formatting

File: homework/Stupns/homework10.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+def tag(name='<div>', name2=''):
+    def decorator(func):
+        def wrapper(*args2):
+            try:
+                if name2 == '':
```
Comment (aodarc) [inline homework/Stupns/homework10.py]: skip second argument

File: homework/Stupns/homework10.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+                    new_name = str(name).replace('<', '</')
+                    return name + func(*args2) + new_name
```
Comment (aodarc) [inline homework/Stupns/homework10.py]: do not concatenate string using plus (+)

**Review by aodarc — CHANGES_REQUESTED** [review]

**Pull request closed without merging**